File: 001/change.rb
def changeCount(amount)
    amountInPennies = (amount * 100).to_i
    quarterCount = 0
    dimeCount = 0
    nickleCount = 0
    pennyCount = 0
    remainder = 0
    
    quarterCount = amountInPennies / 25
    remainder = amountInPennies % 25
    dimeCount = remainder / 10
    remainder = remainder % 10
    nickleCount = remainder / 5
    pennyCount = remainder % 5
    
    Hash["Quarters: " => quarterCount, "Dimes: " => dimeCount, "Nickles: " => nickleCount, "Pennies" => pennyCount]
end

testAmount1=5.00
expectedResult1 = Hash["Quarters: " => 20, "Dimes: " => 0, "Nickles: " => 0, "Pennies" => 0]
result1 = changeCount(testAmount1)
puts result1 == expectedResult1

testAmount2=10.24
expectedResult2 = Hash["Quarters: " => 40, "Dimes: " => 2, "Nickles: " => 0, "Pennies" => 4]
result2 = changeCount(testAmount2)
puts result2 == expectedResult2

testAmount3=0.99
expectedResult3 = Hash["Quarters: " => 3, "Dimes: " => 2, "Nickles: " => 0, "Pennies" => 4]
result3 = changeCount(testAmount3)
puts result3 == expectedResult3

File: 001/change.js
var testAmount01 = 5.00;
var testAmount02 = 10.24;
var testAmount03 = 0.99;
var quarters = 0;
var quartersRemainder = 0;
var dimes = 0;
var dimesRemainder = 0;
var nickles = 0;
var nicklesRemainder = 0;
var pennies = 0;
var penniesRemainder = 0;


function changeCalculator(amount){
    var timesHundred = amount*100;
    quarters = Math.floor(timesHundred / 25);
    quartersRemainder = timesHundred % 25;
    
    if (quartersRemainder > 0){
        dimes = Math.floor(quartersRemainder / 10);
        dimesRemainder = quartersRemainder % 10;
    }
    
    if (dimesRemainder > 0){
        nickles = Math.floor(dimesRemainder / 5);
        nicklesRemainder = dimesRemainder % 5;
    }
    
    pennies = Math.floor(nicklesRemainder);
    
    var result = { 
        "Quarters:": quarters, 
        "Dimes:": dimes,
        "Nickles:": nickles,
        "Pennies:": pennies
    };
    
    return result;
}

// Tests

var result1 = changeCalculator(testAmount01);
console.log(result1);
var expected1 = { 
        "Quarters:": 20, 
        "Dimes:": 0,
        "Nickles:": 0,
        "Pennies:": 0
    };
console.log(expected1);
   
var result2 = changeCalculator(testAmount02);
var expected2 = { 
        "Quarters:": 40, 
        "Dimes:": 2,
        "Nickles:": 0,
        "Pennies:": 4
    };
var result3 = changeCalculator(testAmount03);
var expected3 = { 
        "Quarters:": 3, 
        "Dimes:": 2,
        "Nickles:": 1,
        "Pennies:": 4
    };

if (result1.toString() === expected1.toString()){
    console.log("Test #1 Passed!");
}
else {
    console.log("Test #1 Failed!");
    console.log("Tried with " + testAmount01);
    console.log("Expected to get: " + expected1);
    console.log("Instead, got: " + result1);
}

console.log("\nMoving onto Test #2...\n");
/*
if (result2.sort().toString() === [0, 8, 9, 9].toString()){
    console.log("Test #2 Passed!");
}
else {
    console.log("Test #2 Failed!");
    console.log("Tried with [0, 10, 10, 9, 9, 8]");
    console.log("Expected to get: 0,9,9,8");
    console.log("Instead, got: " + result2);
}*/